ACPs/176-dynamic-evm-gas-limit-and-price-discovery-updates/README.md

```diff
@@ -0,0 +1,157 @@
+
+Let $q$ be a non-negative integer that is initialized to 0 upon activation of this mechanism. Let the target gas consumption per second be expressed as:
+
+$$T = P \cdot e^{q/D}$$
+
+where $P$ is the global minimum allowed target gas consumption rate for the network, and $D$ is a constant that helps control the rate of change of the target gas consumption.
+
+After the execution of transactions in block $b$, the values $q$ can be increased or decreased up to $Q$. It must be the case that $q_{b-1} - Q \leq q_{b} \leq q_{b-1} + Q$, or block $b$ is considered invalid. The amount by which $q$ changes after executing block $b$ is specified by the block builder.
+
+Block builders (i.e. validators), may set their desired value for $T$ (i.e. their desired gas consumption rate) in their configuration, and their desired value for $q$ can then be calculated as:
+
+$$q_{desired} = D \cdot ln(\frac{T}{P})$$
+
+Note that since $q_{desired}$ is only used locally and can be different for each node, it is safe for implementations to approximate the value of $ln(\frac{T}{P})$, and round the resulting value to the nearest integer.
+
+When building a block, builders can calculate their next preferred value for $q$ according to:
+
+  ```python
+  # Calculates a node's new desired value for q given for a given block
+  def calc_next_q(q_current: int, q_desired: int, max_change: int) -> int:
+    if q_desired > q_current:
+        return q_current + min(q_desired - q_current, max_change)
+    else:
+        return q_current - min(q_current - q_desired, max_change)
+  ```
+
+As $q$ is updated after the execution of transactions within the block, $T$ is also updated such that $T = P \cdot e^{q/D}$ at all times. As the value of $T$ adjusts, the value of $R$ (capacity added per second) is also updated such that:
+
+$$R = T \cdot 2$$
+
+This ensures that the gas price remains equally reactive in either direction depending on how far the actual gas consumption rate is from the target, whether it is above or below.
+
+The value of $C$ must also adjust proportionately, so we set:
+
+$$C = R \cdot 10$$
+
+This means that the maximum stored gas capacity would be reached after 10 seconds where no blocks have been accepted.
+
+In order to keep roughly constant the time it takes for the gas price to double at sustained maximum network capacity usage, the value of $K$ used in the gas price determination mechanism must be updated proportionatly to $T$. When updating $T$ from $T_{n}$ to $T_{n+1}$, we update $K$ such that:
+
+$$K_{n+1} = K_{n} \cdot \frac{T_{n+1}}{T_{n}}$$
+
+In order to have the gas price not be directly impacted by the change in $K$, we also update $x$ proportionatly. When updating $x$ after executing a block, instead of setting $x = x + G$ as specified in ACP-103, we set:
```

Comment:
```suggestion
In order to have the gas price not be directly impacted by the change in $K$, we also update $x$ (excess gas consumption) proportionally. When updating $x$ after executing a block, instead of setting $x = x + G$ as specified in ACP-103, we set:
```